k6/generic/change_config_maps.js

```diff
@@ -1,70 +1,467 @@
+      if (body === undefined || body.pagination === undefined || body.pagination.partial === undefined || body.pagination.next === undefined) {
+        break
+      }
+      nextUrl = body.pagination.next
+    } catch (e) {
+      if (e instanceof SyntaxError) {
+        fail("Response body does not parse as JSON: " + res.body)
+      }
+      throw e
+    }
+
+    pause()
+  }
+}
+
+// export function eventsPerSecond() {
+//   // Change Events per second
+//   console.log("# EVENTS")
+//   console.log(changeEvents)
+//   const currentRate = changeEvents / ((Date.now() - exec.test.startTime) / 1000);
+//   console.log("EVENTS/SECOND")
+//   console.log(Math.round(currentRate))
+//   churnOpsRate.add(Math.round(currentRate));
+// }
+
+export function duringChurnDiagnostics(cookies) {
+  console.log('=== Collecting DURING-CHURN diagnostics ===');
+
+  const start = Date.now();
+  const resourceCounts = collectResourceCounts(cookies, null);
+  const apiTimings = collectAPITimings(cookies, null);
+  const duration = Date.now() - start;
+
+  // Record timing for diagnostic collection overhead
+  diagnosticsDuringTrend.add(duration);
+
+  console.log(`During-churn diagnostics collected in ${duration}ms, current rate: ${currentRate.toFixed(2)} ops/s`);
+}
+
+export function postChurnDiagnostics(cookies) {
+  console.log('=== Collecting POST-CHURN diagnostics ===');
+
+  // Wait a bit for operations to settle
+  sleep(5);
+
+  const resourceCounts = collectResourceCounts(cookies, null);
+  const apiTimings = collectAPITimings(cookies, null);
+
+  // Record final metrics
+  diagnosticsUtil.metrics.forEach(({ key, gauge }) => {
+    const count = resourceCounts[key]?.totalCount;
+    if (count != null) {
+      diagnosticsAfterGauge.add(count, { resource: key });
+    }
+  });
+
+  console.log('Post-churn diagnostics collected');
+}
+
+function collectResourceCounts(cookies, tags) {
+  console.log('Collecting resource counts...');
+  const resourceCountsRaw = diagnosticsUtil.getLocalClusterResourceCounts(baseUrl, cookies);
+  const resourceCounts = diagnosticsUtil.processResourceCounts(resourceCountsRaw);
+
+  diagnosticsUtil.metrics.forEach(({ key, label, gauge }) => {
+    const count = resourceCounts[key]?.totalCount;
+    if (count != null) {
+      gauge.add(count, tags);
+      console.log(`Total ${label}: ${count}`);
+    }
+  });
+
+  return resourceCounts;
+}
+
+function collectAPITimings(cookies, tags) {
+  console.log('Collecting API response timings...');
+
+  const timings = diagnosticsUtil.processResourceTimings(baseUrl, cookies);
+
+  diagnosticsUtil.metrics.forEach(({ key, trend }) => {
+    const time = timings[key];
+    if (time != null) {
+      trend.add(time, tags);
+    }
+  });
+
+  const slowest = Object.entries(timings)
+    .sort(([, a], [, b]) => b - a)
+    .slice(0, 5);
+
+  console.log('Top 5 slowest APIs:');
+  slowest.forEach(([api, time], idx) => {
+    console.log(`  ${idx + 1}. ${api}: ${time.toFixed(2)}ms`);
+  });
```

Comment:
How are you consuming data from this script?

Commandline or via Mimir + Grafana?